recipes/frontend-linting-setup.md

```diff
@@ -0,0 +1,61 @@
+# Frontend Linting Setup
+
+---
+Some notes:
+- Using eslint to catch errors/bad patterns. Config file: `/src/frontend/.eslintrc.js`
+- Using prettier to format code. Config file: `/src/frontend/.prettierrc`
+- Custom git hooks defined in `/bin/hooks/`. This formats and lints the code prior to committing.
+
+---
+
+## Git Commit Hook Setup:
+
+Get into hooks directory:
+```
+cd .git/hooks
+```
+Sync custom hook with git: 
+```
+ln -s ../../bin/git_hooks/pre-commit pre-commit
+```
+Go back to project root and make hook executable:
+```
+cd ..; cd ..; chmod +x bin/git_hooks/pre-commit
+```
+Ensure it works by running
+```
+ls -l .git/hooks
+```
+You should see pre-commit hook listed there.
+
+
+Now a couple more files we need to make executable. 
+From root of project run:
+```
+ chmod +x ./bin/linting/format_frontend.sh  
+```
+```
+ chmod +x ./bin/linting/lint_frontend.sh  
+```
```

Comment:
Hmm, this should be handled by `make` (aka `bin/fresh_install.sh`, specifically `bin/create_git_hooks.sh`) -- seems not necessary to include in the recipe, assuming the script handles things for you?

```sh
for f in ./bin/git_hooks/*
do
    mkdir -p .git/hooks/

    # get filename from full path
    filename=$(basename -- "$f")

    echo "making git hook -> ${f} into .git/hooks/${filename}"
    ln -sf "../../bin/git_hooks/${filename}" ".git/hooks/${filename}"
done
```

Reply:
Done